music21/stream/tools.py

```diff
@@ -0,0 +1,170 @@
+    >>> s.append(note.Note('F'))
+    >>> s.append(note.Note('G'))
+
+    >>> s.append(meter.TimeSignature('2/4'))
+    >>> s.append(key.KeySignature(-5))
+    >>> s.append(clef.BassClef())
+
+    >>> s.append(note.Note('A'))
+    >>> s.append(note.Note('B'))
+    >>> s.append(note.Note('C5'))
+
+    Now we'll make it into a proper part with measures and see
+    how it looks in its original, unaltered form:
+
+    >>> s = s.makeMeasures()
+    >>> s.show('t')
+    {0.0} <music21.stream.Measure 1 offset=0.0>
+        {0.0} <music21.clef.TrebleClef>
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.note.Note C>
+        {2.0} <music21.note.Note D>
+    {3.0} <music21.stream.Measure 2 offset=3.0>
+        {0.0} <music21.clef.TrebleClef>
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note E>
+        {1.0} <music21.note.Note F>
+        {2.0} <music21.note.Note G>
+    {6.0} <music21.stream.Measure 3 offset=6.0>
+        {0.0} <music21.clef.BassClef>
+        {0.0} <music21.key.KeySignature of 5 flats>
+        {0.0} <music21.meter.TimeSignature 2/4>
+        {0.0} <music21.note.Note A>
+        {1.0} <music21.note.Note B>
+    {8.0} <music21.stream.Measure 4 offset=8.0>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.bar.Barline type=final>
+
+    Calling removeDuplicates should remove the duplicates
+    even with those changes now stored within measures,
+    not directly on the part.
+    Specifically, in our example,
+    the duplicates entries are removed from measure 2
+    and the actual changes in measure 3 remain.
+
+    >>> s = stream.tools.removeDuplicates(s)
+    >>> s.show('t')
+    {0.0} <music21.stream.Measure 1 offset=0.0>
+        {0.0} <music21.clef.TrebleClef>
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.note.Note C>
+        {2.0} <music21.note.Note D>
+    {3.0} <music21.stream.Measure 2 offset=3.0>
+        {0.0} <music21.note.Note E>
+        {1.0} <music21.note.Note F>
+        {2.0} <music21.note.Note G>
+    {6.0} <music21.stream.Measure 3 offset=6.0>
+        {0.0} <music21.clef.BassClef>
+        {0.0} <music21.key.KeySignature of 5 flats>
+        {0.0} <music21.meter.TimeSignature 2/4>
+        {0.0} <music21.note.Note A>
+        {1.0} <music21.note.Note B>
+    {8.0} <music21.stream.Measure 4 offset=8.0>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.bar.Barline type=final>
+
+    '''
+
+    supportedClasses = [meter.TimeSignature, key.KeySignature, clef.Clef]
+    if classesToRemove is None:
+        classesToRemove = supportedClasses
+
+    listOfObjectsToRemove = []
+
+    for thisClass in classesToRemove:
+
+        if thisClass not in supportedClasses:
```

Comment:
We can liberalize a bit to allow for subclassing. This would be particularly helpful for providing specific clefs.

```suggestion
        if not any(issubclass(thisClass, supportedClass) for supportedClass in supportedClasses):
```